types/bitcoinjs-lib/index.d.ts

```diff
@@ -1,114 +1,234 @@
+
     static makeRandom(options: any): ECPair;
+
 }
+
 export class ECSignature {
     constructor(r: any, s: any, ...args: any[]);
+
     toCompact(i: any, compressed: any): any;
+
     toDER(): any;
+
     toScriptSignature(hashType: any): any;
-    static fromDER(buffer: any): ECSignature;
-    static parseCompact(buffer: any): ECSignature;
-    static parseScriptSignature(buffer: any): ECSignature;
+
+    static fromDER(buffer: Buffer): any;
+
+    static parseCompact(buffer: Buffer): any;
+
+    static parseScriptSignature(buffer: Buffer): any;
+
 }
+
 export class HDNode {
-    constructor(keyPair: any, chainCode: any, ...args: any[]);
-    derive(index: any): any;
-    deriveHardened(index: any): any;
-    derivePath(path: any): any;
+    constructor(keyPair: ECPair, chainCode: any, ...args: any[]);
+
+    keyPair: ECPair // dl added manually TODO check if correct
+
+    derive(index: number): HDNode;
+
+    deriveHardened(index: number): HDNode;
+
+    derivePath(path: string): HDNode;
+
     getAddress(): string;
-    getFingerprint(): any;
-    getIdentifier(): any;
-    getNetwork(): Network;
-    getPublicKeyBuffer(): any;
-    isNeutered(): any;
-    neutered(): any;
-    sign(hash: any): any;
-    toBase58(__isPrivate: any): any;
-    toString(__isPrivate: any): any;
-    verify(hash: any, signature: any): any;
+
+    getFingerprint(): Buffer;
+
+    getIdentifier(): Buffer;
+
+    getNetwork(): any;
+
+    getPublicKeyBuffer(): Buffer;
+
+    isNeutered(): boolean;
+
+    neutered(): HDNode;
+
+    sign(hash: Buffer): ECSignature;
+
+    toBase58(__isPrivate?: any): string;
+
+    verify(hash: Buffer, signature: any): Buffer;
+
     static HIGHEST_BIT: number;
+
     static LENGTH: number;
-    static fromBase58(string: any, networks: Network[]): HDNode;
-    static fromSeedBuffer(seed: any, network: Network, ...args: any[]): HDNode;
-    static fromSeedHex(hex: any, network: Network): HDNode;
+
+    static fromBase58(string: string, networks?: any): HDNode;
+
+    static fromSeedBuffer(seed: Buffer, network: Network, ...args: any[]): HDNode;
+
+    static fromSeedHex(hex: string, network: Network): HDNode;
+
 }
+
 export class Transaction {
+    version: number;
+    locktime: number;
+    ins: Array<Input>;
+    outs: Array<Output>;
     constructor();
-    addInput(hash: any, index: any, sequence: any, scriptSig: any, ...args: any[]): any;
+
+    addInput(hash: Buffer, index: number, sequence: any, scriptSig: Buffer, ...args: any[]): any;
+
     addOutput(scriptPubKey: any, value: any, ...args: any[]): any;
-    byteLength(): number;
+
+    byteLength(): any;
+
     clone(): Transaction;
-    getHash(): any;
+
+    getHash(): Buffer;
+
     getId(): any;
-    hashForSignature(inIndex: any, prevOutScript: any, hashType: any, ...args: any[]): any;
+
+    hasWitnesses(): any;
+
+    hashForSignature(inindex: number, prevOutScript: any, hashType: any, ...args: any[]): any;
```

Comment:
Looks like this should be:
```ts
hashForSignature(inIndex: number, prevOutScript: Buffer, hashType: number): Buffer;
```
The parameter `...args` is unused (exists only because `arguments` is used in function for type checking).